src/assets/htmx/htmx.md

```diff
@@ -0,0 +1,374 @@
+Now on the server
+
+```go
+func save(w http.ResponseWriter, r *http.Request) {
+    r.ParseForm()
+    storeTODO(r.FormValue("task")) // save to database
+    tastList := "<div>" + r.FormValue("task") + " </div>"
+    io.WriteString(w, tastList)
+}
+
+func main() {
+    ...
+    http.HandleFunc("/save", save)
+    ...
+}
+```
+
+![demo](src/assets/htmx/wmTHPYD.gif)
+
+
+In the above html, the form will send a post request to `/save` endpoint and htmx will replace the content in the div with the id `todo-list` but the response will replace every thing the target element i.e. element with id `todo-list` will be replaced.
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+</div>
+```
+
+After adding another
+
+```html
+<div id="todo-list">
+  <div>item 2</div>
+</div>
+```
+
+Instead, We want to only append the new todo item after the last todo item. so we use the attribute `hx-swap` to let htmx know where we want the response to render. The `beforeend` value will append the html response before the end of the targeted element.
+
+After the swap attribute
+
+```html
+<div id="app" class="container">
+  <div id="todo-list"></div>
+  <form
+    id="myForm"
+    hx-post="/save"
+    hx-trigger="submit"
+    hx-target="#todo-list"
+    hx-swap="beforeend"
+  >
+    <input type="text" name="task" required />
+    <input type="submit" value="save" />
+  </form>
+</div>
+```
+
+(adding item 1)
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+</div>
+```
+
+After adding another item (item 2)
+
+```html
+<div id="todo-list">
+  <div>item 1</div>
+  <div>item 2</div>
+</div>
+```
+
+
+
+
+But on page load (e.g. if we refresh the page) the list of tasks will not be shown.
+To make that work, we need to make a `GET` request.
+
+```html
+<div id="todo-list" hx-get="/taskList" hx-trigger="load"></div>
+```
+
+In the above html, I have added `hx-get` with value `/tasklist` and `hx-trigger` with value 'load'. The `load` value specifies when the request is sent in this case it's when the html file is loaded.
+
+```go
+func listTask(w http.ResponseWriter, r *http.Request) {
+    tastList := getTODO() // gets data from database and returns html
+    io.WriteString(w, tastList)
+}
+func main() {
+    ...
+    http.HandleFunc("/save", listTask)
+    ...
+}
+```
+![demo](src/assets/htmx/yMVvdP3.gif)
+
+Now the list is shown even when the page is refreshed.
+
+For edit let's try some thing different. Previously we did all our operations in a single page for edit we will be navigating to another page without refreshing the wholepage unlike normal html page navigation using an anchor tag.
+
+First let's make clicking on the todo item go to a different page that will contain form for updating the todo item.
+For that, we need to get an anchor tag instead of a div with the todo text inside a div
+
+
+
+In the server i will get the task and return the proper form for updating that task.
+
+```go
+func main() {
+    ...
+    http.HandleFunc("/task/", indivisualTask)
```

Comment:
that's a rather unique name for the variable... maybe individualTask or something similar is more intuitive for the reader...